composer/trainer/trainer.py

```diff
@@ -705,15 +733,26 @@ def _train_batch_inner(self, microbatches: Sequence[Batch]):
                 for loss in ensure_tuple(state.loss):
                     loss.backward(create_graph=self.backwards_create_graph)
 
+                for loss in ensure_tuple(state.loss):
```

Comment:
Should this be before the loss is scaled? Or was that a bug all along?

Reply:
DeepSpeed handles loss scaling on its own, but doesn't do it in place. I've since updated the code to reflect the awkward reality that for non-DeepSpeed, we need to do this scaling before we call `loss.backward`, and for DeepSpeed, we need to scale after the backward pass.